sale_import_base/components/importer_sale_channel.py

```diff
@@ -165,16 +107,27 @@ def _prepare_partner(self, data):
             "phone": data.get("phone"),
             "mobile": data.get("mobile"),
         }
-        if data.get("state_code"):
-            state = self.env["res.country.state"].search(
-                [("code", "=", data["state_code"])]
-            )
-            result["state_id"] = state.id
         if data.get("country_code"):
             country = self.env["res.country"].search(
                 [("code", "=", data["country_code"])]
             )
+            if len(country.ids) != 1:
+                raise ValidationError(
+                    _("Could not determine one country from country code")
+                )
             result["country_id"] = country.id
+        if data.get("state_code"):
+            state = self.env["res.country.state"].search(
+                [
+                    ("code", "=", data["state_code"]),
+                    ("country_id", "in", [result.get("country_id")]),
```

Comment:
("country_id", "=", result.get("country_id")),

Reply:
this whole part is made conditional on there being a country